File: dayXXX/aoc.py
import re
import sys
import os
sys.path.insert(1, os.path.join(sys.path[0], '..'))
import utils.utils as ut


def part1():
    print('Part1:')


def part2():
    print('Part2:')


def main():
    if len(sys.argv) < 2:
        print('Provide input file name')
        exit(-1)

    #lines = ut.read_groups(sys.argv[1])
    lines = ut.read_lines(sys.argv[1])
    for line in lines:
        print(line)

    part1()
    part2()   


if __name__ == '__main__':
    main()

File: day9/aoc.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <map>
#include <regex>
#include <filesystem>
#include "utils.h"


using namespace std;


void part1()
{
    cout << "Part1: " << '\n';
}


void part2()
{
    cout << "Part2: " << '\n';
}


void solution(std::istream& is)
{
    auto lines = aoc::read_lines(is);
    //auto lines = aoc::read_groups(is);

    // Place holder for creating data structure.
    for (const auto& line: lines)
    {
        cout << line << "\n\n";
    }

    part1();
    part2();
}


int main(int argc, char** argv)
{
    if (argc < 2)
    {
        std::cout << "Provide input file name\n";
        return -1;
    }
    if (!std::filesystem::is_regular_file(argv[1]))
    {
        std::cout << "Provide valid input file name\n";
        return -1;
    }

    std::ifstream is{argv[1]};
    solution(is);
}
